### PR TITLE
Avoid potential memory leak when dropping mqtt clients

### PR DESCRIPTION
**Pull request by HalfVoxel**
- Improve docs for mqtt/client
- Improve docs for private/zerocopy
- Avoid potential memory leak when dropping mqtt clients

### CODE REVIEW

**Comment by ivmarkov** [comment]
Ping me when this is ready for review.

**Comment by HalfVoxel** [comment]
Should be ready now.

File: src/private/unblocker.rs [hunk 1]
```diff
@@ -73,10 +79,22 @@ where
...
+        }
+        unsafe {
+            // FreeRTOS tasks must delete themselves. Returning from the task function
+            // without deleting the task will cause a crash.
+            // This function will immediately stop running. This means we have to ensure
+            // that all memory is freed before calling vTaskDelete.
+            // Fortunately, this function does not manage any memory at this point.
+            // The scope above guarantees that all relevant variables are dropped at this point.
+            // See https://www.freertos.org/implementing-a-FreeRTOS-task.html
+            task::destroy(std::ptr::null_mut());
```
Comment (Vollbrecht) [inline src/private/unblocker.rs:96]: replace `std` with `core` please. That will also make CI happy.

Comment (HalfVoxel) [reply src/private/unblocker.rs]: Fixed

**Pull request merged**